Subject:       Fixed version checking with 2021.2.0 in FindTBB.cmake
Author:        Orson Teodoro <orsonteodoro@hotmail.com>
Patsh status:  Testing
Date:          Mon Jun 28 03:48:23 AM PDT 2021 (Unix timestamp: 1624877303)

Copyright notice of cmake/modules/FindTBB.cmake:

# Obtained from Justus Calvin:
# https://github.com/justusc/FindTBB/blob/25ecdea817b3af4a26d74ddcd439642dbd706acb/FindTBB.cmake
#
# With the following modifications:
# * Move the "tbb" imported library target into a namespace as "TBB::tbb" to
#   conform to modern CMake conventions.
# * Append "lib" as a library path suffix on all platforms.
#
# The MIT License (MIT)
#
# Copyright (c) 2015 Justus Calvin
# 
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
# 
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.


diff -urp USD-21.05.orig/cmake/modules/FindTBB.cmake USD-21.05/cmake/modules/FindTBB.cmake
--- USD-21.05.orig/cmake/modules/FindTBB.cmake	2021-04-12 15:09:41.000000000 -0700
+++ USD-21.05/cmake/modules/FindTBB.cmake	2021-06-28 03:42:38.216476894 -0700
@@ -197,7 +197,11 @@ if(NOT TBB_FOUND)
   ##################################
 
   if(TBB_INCLUDE_DIRS)
-    file(READ "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h" _tbb_version_file)
+    if (EXISTS "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h")					# oteodoro: added
+      file(READ "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h" _tbb_version_file)		# oteodoro: indent line
+    else()										# oteodoro: added line
+      file(READ "${TBB_INCLUDE_DIRS}/oneapi/tbb/version.h" _tbb_version_file)		# oteodoro: added line
+    endif()										# oteodoro: added line
     string(REGEX REPLACE ".*#define TBB_VERSION_MAJOR ([0-9]+).*" "\\1"
         TBB_VERSION_MAJOR "${_tbb_version_file}")
     string(REGEX REPLACE ".*#define TBB_VERSION_MINOR ([0-9]+).*" "\\1"
